gen/nvml/lib.go

```diff
@@ -77,10 +78,14 @@ var newDynamicLibrary = func(path string, flags int) dynamicLibrary {
 
 // load initializes the library and updates the versioned symbols.
 // Multiple calls to an already loaded library will return without error.
-func (l *library) load() error {
+func (l *library) load() (rerr error) {
 	l.Lock()
 	defer l.Unlock()
-	if l.dl != nil {
+
+	l.refcount.Inc()
```

Comment:
Question: Does something like:
```
defer l.refcount.IncOnNoError(rerror)
```
make sense here. (adjusting the `l.refcount > 1` of course).

Also, are we assured that this defer would execute before the unlock?